lib/core/renderer.js

```diff
@@ -306,7 +306,7 @@ export default class Renderer {
     const spa = context.spa || (context.res && context.res.spa)
     const ENV = this.options.env
 
-    if (this.noSSR || spa) {
+    const spaRenderer = async () => {
```

Comment:
Good, refactor idea. But why it should be async? Also, can we rename it to `renderSpa`?